xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/ModelBridge.java

```diff
@@ -60,4 +61,14 @@
      * @throws NotificationException if an error occurs
      */
     void setStartDateForUser(DocumentReference userReference, Date startDate) throws NotificationException;
+
+    /**
+     * Return the notification events of a given document.
+     *
+     * @param documentReference the document reference
+     * @return the list of applied notifications events
+     * @throws NotificationException if an error occurs
+     */
+    List<PageNotificationEventDescriptor> getPageNotificationEventDescriptors(DocumentReference documentReference)
```

Comment:
Why a list? A page can contains several descriptors? Technically, I know it is possible, but I'm not sure we should encourage this.